starknet-foundry/crates/forge/src/lib.rs

```diff
@@ -70,27 +70,36 @@ struct TestsFromFile {
 }
 
 fn collect_tests_from_directory(
-    input_path: &Utf8PathBuf,
+    package_path: &Utf8PathBuf,
+    lib_path: &Utf8PathBuf,
     linked_libraries: &Option<Vec<LinkedLibrary>>,
     corelib_path: Option<&Utf8PathBuf>,
     runner_config: &RunnerConfig,
 ) -> Result<Vec<TestsFromFile>> {
-    let test_files = find_cairo_files_in_directory(input_path)?;
+    let test_files = find_cairo_root_files_in_directory(package_path, lib_path)?;
     internal_collect_tests(
-        input_path,
+        package_path,
         linked_libraries,
         &test_files,
         corelib_path,
         runner_config,
     )
 }
 
-fn find_cairo_files_in_directory(input_path: &Utf8PathBuf) -> Result<Vec<Utf8PathBuf>> {
-    let mut test_files: Vec<Utf8PathBuf> = vec![];
+fn find_cairo_root_files_in_directory(
+    package_path: &Utf8PathBuf,
+    lib_path: &Utf8PathBuf,
+) -> Result<Vec<Utf8PathBuf>> {
+    let mut test_files: Vec<Utf8PathBuf> = vec![lib_path.clone()];
+    let src_path = lib_path.parent().unwrap();
```

Comment:
```suggestion
    let src_path = lib_path.parent().context("Failed to get parent directory of package")?;
```
Or something like that